use-cases/inferencing/serving-with-vllm/README.md

```diff
@@ -0,0 +1,513 @@
+  
+  ```sh
+  MODEL-ID=<your-model-id>
+  sed -i -e "s|_MODEL-ID_|${MODEL-ID}|" manifests/gradio.yaml
+  ```
+
+  ```sh
+  sed -i -e "s|_NAMESPACE_|${NAMESPACE}|" manifests/gradio.yaml
+  kubectl apply -f manifests/gradio.yaml
+  ```
+
+### Production Metrics
+
+vLLM exposes a number of metrics that can be used to monitor the health of the system. These metrics are exposed via the `/metrics` endpoint on the vLLM OpenAI compatible API server.
+
+  ```sh
+  curl http://localhost:8000/metrics
+  ```
+
+### View vLLM serving metrics for your model on GKE
+
+You can configure monitoring of the metrics above using the [pod monitoring](https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-managed#gmp-pod-monitoring)
+
+  ```sh
+  kubectl apply -f manifests/pod_monitoring.yaml
+  ```
+
+### Create a dashboard for Cloud Monitoring to view vLLM metrics
+
+Cloud Monitoring provides an [importer](https://cloud.google.com/monitoring/dashboards/import-grafana-dashboards) that you can use to import dashboard files in the Grafana JSON format into Cloud Monitoring
+
+ Clone github repository
+
+  ```sh
+  git clone https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples
+  ```
+
+ Change to the directory for the dashboard importer:
+
+  ```sh
+  cd monitoring-dashboard-samples/scripts/dashboard-importer
+  ```
+
+The dashboard importer includes the following scripts:
+
+- import.sh, which converts dashboards and optionally uploads the converted dashboards to Cloud Monitoring.
+- upload.sh, which uploads the converted dashboards—or any Monitoring dashboards—to Cloud Monitoring. The import.sh script calls this script to do the upload.
+
+ Import the dashboard
+
+  ```sh
+  ./import.sh ./configs/grafana.json ${PROJECT_ID}
+  ```
+
+  When you use the import.sh script, you must specify the location of the Grafana dashboards to convert. The importer creates a directory that contains the converted dashboards and other information.
+
+
+### Run Batch inference on GKE
+
+Once a model has completed fine-tuning and is deployed on GKE , its ready to run batch Inference pipeline.
+In this example batch inference pipeline, we would first send prompts to the hosted fine-tuned model and then validate the results based on ground truth.
+
+#### Prepare your environment
+
+
+Set env variables.
+
+```
+PROJECT_ID=<your-project-id>
+PROJECT_NUMBER=$(gcloud projects describe ${PROJECT_ID} --format="value(projectNumber)")
+CLUSTER_NAME=<your-gke-cluster>
+NAMESPACE=ml-serve
+MODEL_PATH=<your-model-path>
+BUCKET="<your dataset bucket name>"
+DATASET_OUTPUT_PATH=""
+ENDPOINT=<your-endpoint> # eg "http://vllm-openai:8000/v1/chat/completions"
+KSA=<k8s-service-account> # Service account with work-load identity enabled
+PREDICTIONS_FILE=<predictions.txt> # Look for sample example-prediction.text for expected results
+```
+
+Create Service account.
+
+```
+NAMESPACE=ml-serve
+kubectl create sa ${KSA} -n ${NAMESPACE}
```

Comment:
need to note, platform will create SA/bucket permissions

Reply:
We need a separate namespace for hosting the model for security and functionality reasons. In my opinion, enterprises would prefer to host the model at least in a separate namespace and hence restrict all the resource access to the serving namespace only. This is a separate name space than ml-team

Reply:
thanks!